packages/blaze/view.js

```diff
@@ -352,13 +352,16 @@ Blaze._materializeView = function (view, parentView, _workStack, _intoArray) {
       view._isInRender = false;
 
       if (! c.firstRun) {
+        // added by ramezrafla@zegenie to skip affecting DOM for unchanged content
+        if (Blaze._deepCompare(lastHtmljs,htmljs)) {
```

Comment:
So I looked into this change [using this example](https://github.com/mitar/test-blaze-array) and I could not get it to in fact come to this `if (! c.firstRun) {` condition with any other view than `lookup:something`.

If I did:

```
Collection.update({}, {$set: {value: 'a'}})
```

Change was from `aa` to `a0`.

And then if I did:

```
Collection.update({}, {$set: {value: 'a', test: 'b'}})
```

Change was from `aa` to `aa`.

So, I think that `Blaze._isContentEqual` moving up is a great idea for cases like above, where I am not minimizing my fields selector well in Minimongo. But I am not sure if there is any benefit of changing the function to deep-comparison? Can you please adapt my example to the case where you get something else than those simple values `Blaze._isContentEqual` is already checking? I could imagine only very extreme cases where you would be using `{{value}}` where `value` is not a simple variable, but an object, and you are using its stringified value to be shown in the template. But is anyone really doing that?